Fairfax Australia

```diff
@@ -0,0 +1,708 @@
+	this._filters = new Array();
+
+	this.text = function() {
+		var filter = function(n) {
+			if (typeof n === 'object' && n.textContent) { return n.textContent; }
+			else { return n; }
+		};
+		this.addFilter(filter);
+		return this;
+	};
+
+	this.sub = function(xpath) {
+		var filter = function(n, doc) {
+			var result = doc.evaluate(xpath, n, null, XPathResult.ANY_TYPE, null);
+			if (result) {
+				return result.iterateNext();
+			} else {
+				return undefined;               
+			}
+		};
+		this.addFilter(filter);
+		return this;
+	};
+
+	this.evaluate = function (doc) {
+		var res = doc.evaluate(this._xpath, doc, null, XPathResult.ANY_TYPE, null);
+		var resultType = res.resultType;
+		var a = new Array();
+		if (resultType == XPathResult.STRING_TYPE) {
+			a.push(res.stringValue);
+		} else if (resultType == XPathResult.BOOLEAN_TYPE) {
+			a.push(res.booleanValue);
+		} else if (resultType == XPathResult.NUMBER_TYPE) {
+			a.push(res.numberValue);
+		} else if (resultType == XPathResult.ORDERED_NODE_ITERATOR_TYPE ||
+				   resultType == XPathResult.UNORDERED_NODE_ITERATOR_TYPE) {
+			var x;
+			while ((x = res.iterateNext())) { a.push(x); }
+		} 
+		a = this._applyFilters(a, doc);
+		if (a.length == 0) { return false; }
+		else { return a; }
+	};
+};
+
+FW._Xpath.prototype = new FW._StringMagic();
+
+FW.detectWeb = function (doc, url) {
+	for (var i in FW._scrapers) {
+	var scraper = FW._scrapers[i];
+	var itemType = scraper.evaluateThing(scraper['itemType'], doc, url);
+	var v = scraper.evaluateThing(scraper['detect'], doc, url);
+		if (v.length > 0 && v[0]) {
+		return itemType;
+	}
+	}
+	return undefined;
+};
+
+FW.getScraper = function (doc, url) {
+	var itemType = FW.detectWeb(doc, url);
+	return FW._scrapers.filter(function(s) {
+		return (s.evaluateThing(s['itemType'], doc, url) == itemType)
+		&& (s.evaluateThing(s['detect'], doc, url));
+	})[0];
+};
+
+FW.doWeb = function (doc, url) {
+	var scraper = FW.getScraper(doc, url);
+	scraper.makeItems(doc, url, [], 
+					  function(item, scraper, doc, url) {
+						  scraper.callHook('scraperDone', item, doc, url);
+						  if (!item['title']) {
+							  item['title'] = "";
+						  }
+						  item.complete();
+					  },
+					  function() {
+						  Zotero.done();
+					  });
+	Zotero.wait();
+};
+/*********************** END FRAMEWORK ***********************/
+
+
+function detectWeb(doc, url) { return FW.detectWeb(doc, url); }
+function doWeb(doc, url) { return FW.doWeb(doc, url); }
+
+FW.Scraper({
+itemType	: 'newspaperArticle',
+detect		: FW.Xpath('.//h1[@itemprop="name headline"]'),
+title		: FW.Xpath('.//h1[@itemprop="name headline"]').text().trim(),
+attachments : [ {url: FW.Xpath('//div[@class="article__toolbar"]/ul/li[3]/a').key("href"),
+  title: FW.Xpath('//meta[@name="application-name"]/@content').text().append( " Printable"),
+  type:  "text/html",
+}],
+
+date		: FW.Xpath('//*[@id="content"]/main/article/header/div[1]/ul/li[1]/time').text().trim(),
```

Comment:
this looks very fragile to me -- I'd expect the time element is unique within the article header? so couldn't we just use:
`//*[@id="content"]/main/article/header//time`?